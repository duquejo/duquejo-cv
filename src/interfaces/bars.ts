export interface Bars {
  color: string;
  tag:   string;
  value: number;
}