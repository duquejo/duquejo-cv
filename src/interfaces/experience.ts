
export interface Experience {
  start_date:      string;
  end_date:        string;
  enterprise:      string;
  isRecent:        boolean;
  project:         string;
  role:            string;
  type:            Type;
  resume:          string;
  additional_info: string;
}

export type Sizes = 'small' | 'large' | 'medium';

export enum Type {
  List = 'list',
  Paragraph = 'paragraph',
}