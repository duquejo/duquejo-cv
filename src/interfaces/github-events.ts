
// Generated by https://quicktype.io

export interface Event {
  id:         string;
  type:       EventType;
  actor:      Actor;
  repo:       EventRepo;
  payload:    Payload;
  public:     boolean;
  created_at: string;
}

export interface Actor {
  id:            number;
  login:         string;
  display_login: string;
  gravatar_id:   string;
  url:           string;
  avatar_url:    string;
}

export interface Payload {
  repository_id?: number;
  push_id?:       number;
  size?:          number;
  distinct_size?: number;
  ref?:           null | string;
  head?:          string;
  before?:        string;
  commits?:       Commit[];
  ref_type?:      string;
  master_branch?: string;
  description?:   string;
  pusher_type?:   string;
  action?:        string;
  number?:        number;
  pull_request?:  PullRequest;
}

export interface Commit {
  sha:      string;
  author:   Author;
  message:  string;
  distinct: boolean;
  url:      string;
}

export interface Author {
  email: string;
  name:  string;
}

export interface PullRequest {
  url:                   string;
  id:                    number;
  node_id:               string;
  html_url:              string;
  diff_url:              string;
  patch_url:             string;
  issue_url:             string;
  number:                number;
  state:                 string;
  locked:                boolean;
  title:                 string;
  user:                  User;
  body:                  null;
  created_at:            string;
  updated_at:            string;
  closed_at:             null | string;
  merged_at:             null | string;
  merge_commit_sha:      null | string;
  assignee:              null;
  assignees:             any[];
  requested_reviewers:   any[];
  requested_teams:       any[];
  labels:                any[];
  milestone:             null;
  draft:                 boolean;
  commits_url:           string;
  review_comments_url:   string;
  review_comment_url:    string;
  comments_url:          string;
  statuses_url:          string;
  head:                  Base;
  base:                  Base;
  _links:                Links;
  author_association:    string;
  auto_merge:            null;
  active_lock_reason:    null;
  merged:                boolean;
  mergeable:             null;
  rebaseable:            null;
  mergeable_state:       string;
  merged_by:             User | null;
  comments:              number;
  review_comments:       number;
  maintainer_can_modify: boolean;
  commits:               number;
  additions:             number;
  deletions:             number;
  changed_files:         number;
}

export interface Links {
  self:            Comments;
  html:            Comments;
  issue:           Comments;
  comments:        Comments;
  review_comments: Comments;
  review_comment:  Comments;
  commits:         Comments;
  statuses:        Comments;
}

export interface Comments {
  href: string;
}

export interface Base {
  label: string;
  ref:   string;
  sha:   string;
  user:  User;
  repo:  BaseRepo;
}

export interface BaseRepo {
  id:                          number;
  node_id:                     string;
  name:                        string;
  full_name:                   string;
  private:                     boolean;
  owner:                       User;
  html_url:                    string;
  description:                 string;
  fork:                        boolean;
  url:                         string;
  forks_url:                   string;
  keys_url:                    string;
  collaborators_url:           string;
  teams_url:                   string;
  hooks_url:                   string;
  issue_events_url:            string;
  events_url:                  string;
  assignees_url:               string;
  branches_url:                string;
  tags_url:                    string;
  blobs_url:                   string;
  git_tags_url:                string;
  git_refs_url:                string;
  trees_url:                   string;
  statuses_url:                string;
  languages_url:               string;
  stargazers_url:              string;
  contributors_url:            string;
  subscribers_url:             string;
  subscription_url:            string;
  commits_url:                 string;
  git_commits_url:             string;
  comments_url:                string;
  issue_comment_url:           string;
  contents_url:                string;
  compare_url:                 string;
  merges_url:                  string;
  archive_url:                 string;
  downloads_url:               string;
  issues_url:                  string;
  pulls_url:                   string;
  milestones_url:              string;
  notifications_url:           string;
  labels_url:                  string;
  releases_url:                string;
  deployments_url:             string;
  created_at:                  string;
  updated_at:                  string;
  pushed_at:                   string;
  git_url:                     string;
  ssh_url:                     string;
  clone_url:                   string;
  svn_url:                     string;
  homepage:                    null;
  size:                        number;
  stargazers_count:            number;
  watchers_count:              number;
  language:                    string;
  has_issues:                  boolean;
  has_projects:                boolean;
  has_downloads:               boolean;
  has_wiki:                    boolean;
  has_pages:                   boolean;
  has_discussions:             boolean;
  forks_count:                 number;
  mirror_url:                  null;
  archived:                    boolean;
  disabled:                    boolean;
  open_issues_count:           number;
  license:                     null;
  allow_forking:               boolean;
  is_template:                 boolean;
  web_commit_signoff_required: boolean;
  topics:                      any[];
  visibility:                  string;
  forks:                       number;
  open_issues:                 number;
  watchers:                    number;
  default_branch:              string;
}

export interface User {
  login:               string;
  id:                  number;
  node_id:             string;
  avatar_url:          string;
  gravatar_id:         string;
  url:                 string;
  html_url:            string;
  followers_url:       string;
  following_url:       string;
  gists_url:           string;
  starred_url:         string;
  subscriptions_url:   string;
  organizations_url:   string;
  repos_url:           string;
  events_url:          string;
  received_events_url: string;
  type:                UserType;
  site_admin:          boolean;
}

export enum UserType {
  User = 'User',
}

export interface EventRepo {
  id:   number;
  name: string;
  url:  string;
}

export enum EventType {
  CreateEvent = 'CreateEvent',
  PullRequestEvent = 'PullRequestEvent',
  PushEvent = 'PushEvent',
}
