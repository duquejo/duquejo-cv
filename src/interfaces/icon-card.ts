
export interface IconCard {
  title: string;
  content: string;
  icon: string;
}

export enum Order {
  Col = 'col',
  Row = 'row'
};