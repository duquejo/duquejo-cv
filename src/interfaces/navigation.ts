export type NavigationLayout = 'menu' | 'fixed';