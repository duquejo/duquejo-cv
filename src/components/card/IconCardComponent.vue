
<template>
  <div :class="iconClasses">
    <v-icon :name="props.icon" scale="2" />
    <div class="mt-3 lg:mt-0">
      <h4 v-text="props.title" />
      <p class="text-center lg:text-justify" v-text="props.content" />
    </div>
  </div>
</template>

<script setup lang="ts">
import { computed } from 'vue';
import type { IconCard } from '@/interfaces';
import { Order } from '@/interfaces';

interface Props extends IconCard {
  order?: Order;
  title: string;
  content: string;
  icon: string;
}

const props = withDefaults(defineProps<Props>(), {
  order: Order.Row,
});

const iconClasses = computed(() => ({
  'icon-container': true,
  'flex-col lg:flex-row pl-4 pr-8 text-center lg:text-left': props.order === Order.Row,
  'flex-col px-4 text-center justify-start lg:justify-evenly lg:items-center': props.order === Order.Col,
}));

</script>