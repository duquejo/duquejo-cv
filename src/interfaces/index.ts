import type { Bars } from './bars';
import type { Experience } from './experience';
import type { IconCard } from './icon-card';
import { Type } from './experience';
import { Order } from './icon-card';


export { Bars, Experience, IconCard, Order, Type };