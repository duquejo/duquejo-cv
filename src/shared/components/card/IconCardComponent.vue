
<template>
  <div :class="iconClasses">
    <v-icon :name="props.icon" scale="2" />
    <div class="mt-3 lg:mt-0">
      <h4 v-if="props.title" v-text="props.title" />
      <p v-if="props.content" v-text="props.content" />
    </div>
  </div>
</template>

<script setup lang="ts">
import { computed } from 'vue';
import type { IconCard } from '../../../interfaces';
import { Order } from '../../../interfaces/icon-card';

interface Props extends IconCard {
  order?: Order;
}

const props = withDefaults(defineProps<Props>(), {
  order: Order.Row,
});

const iconClasses = computed(() => ({
  'icon-container': true,
  'flex-col lg:flex-row pl-4 pr-8 text-center lg:text-left': props.order === Order.Row,
  'flex-col px-4 text-center justify-start lg:justify-normal': props.order === Order.Col,
}));

</script>