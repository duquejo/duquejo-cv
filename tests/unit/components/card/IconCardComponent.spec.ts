
import { describe, it, expect } from 'vitest';
import { shallowMount } from '@vue/test-utils';
import IconCardComponent from '../../../../src/components/card/IconCardComponent.vue';
import { Order } from '../../../../src/interfaces/icon-card';


describe('+ IconCardComponent.vue tests', () => {

  const properties = {
    title: 'john',
    content: 'doe',
    icon: 'foo-bar'
  };

  const wrapper = shallowMount(IconCardComponent, {
    props: properties,
  });

  it('Should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Should render a card with it default row configuration', () => {
    expect(wrapper.classes()).toEqual([
      'icon-container',
      'flex-col',
      'lg:flex-row',
      'pl-4',
      'pr-8',
      'text-center',
      'lg:text-left'
    ]);
  });

  it('Should render a column ordered card', () => {
    const wrapperCol = shallowMount(IconCardComponent, {
      props: {
        ...properties,
        order: Order.Col,
      }
    });
    expect(wrapperCol.classes()).toEqual([
      'icon-container',
      'flex-col',
      'px-4',
      'text-center',
      'justify-start',
      'lg:justify-evenly',
      'lg:items-center'
    ]);
  });

  it('Should render the provided props', () => {
    const h4 = wrapper.find('h4');
    const p = wrapper.find('p');
    const icon = wrapper.find('v-icon');

    expect(h4.text()).toBe(properties.title);
    expect(p.text()).toBe(properties.content);
    expect(icon.attributes()).toEqual({
      name: properties.icon,
      scale: expect.any(String),
    });
  });
});