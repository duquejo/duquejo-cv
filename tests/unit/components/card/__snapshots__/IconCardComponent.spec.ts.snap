// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`+ IconCardComponent.vue tests > Should match the snapshot 1`] = `
"<div class="icon-container flex-col lg:flex-row pl-4 pr-8 text-center lg:text-left">
  <v-icon name="foo-bar" scale="2"></v-icon>
  <div class="mt-3 lg:mt-0">
    <h4>john</h4>
    <p class="text-center lg:text-justify">doe</p>
  </div>
</div>"
`;
